src/main/java/de/tum/cit/aet/artemis/core/service/telemetry/TelemetryService.java

```diff
@@ -0,0 +1,86 @@
+package de.tum.cit.aet.artemis.core.service.telemetry;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_SCHEDULING;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.stereotype.Service;
+
+import de.tum.cit.aet.artemis.core.service.ProfileService;
+
+@Service
+@Profile(PROFILE_SCHEDULING)
+public class TelemetryService {
+
+    private static final Logger log = LoggerFactory.getLogger(TelemetryService.class);
+
+    private final ProfileService profileService;
+
+    private final TaskScheduler taskScheduler;
+
+    private final TelemetrySendingService telemetrySendingService;
+
+    private final boolean useTelemetry;
+
+    private final boolean sendAdminDetails;
+
+    private final boolean eurekaEnabled;
+
+    private final long sendingDelay;
+
+    public TelemetryService(ProfileService profileService, TelemetrySendingService telemetrySendingService, @Value("${artemis.telemetry.enabled}") boolean useTelemetry,
+            @Value("${artemis.telemetry.sendAdminDetails}") boolean sendAdminDetails, @Value("${eureka.client.enabled:false}") boolean eurekaEnabled,
+            @Value("${artemis.telemetry.sendingDelay:180}") long sendingDelay) {
+
+        this.profileService = profileService;
+        this.telemetrySendingService = telemetrySendingService;
+        this.taskScheduler = new ThreadPoolTaskScheduler();
+        ((ThreadPoolTaskScheduler) this.taskScheduler).initialize();
```

Comment:
Why does this have to be casted here? Would it be better to set the attribute type to `ThreadPoolTaskScheduler`?

Reply:
That would be indeed better, I'll do that when we (maybe) re-add it in a follow up. For now the multi node node counting is removed